test/functional/page_objects/console_page.ts

```diff
@@ -136,6 +136,9 @@ export class ConsolePageObject extends FtrService {
 
   // Prompt autocomplete window and provide a initial letter of properties to narrow down the results. E.g. 'b' = 'bool'
   public async promptAutocomplete(letter = 'b') {
+    // give browser enough moments to complete events from past UI operations
+    await this.common.sleep(200);
```

Comment:
I'd rather avoid using `sleep` if at all possible. There are other ways to wait for UI loading to compelte, for example `waitFor`.

Reply:
After reading the codes further, I found `evaluateCurrentTokenAfterAChange` function is protected by debounce with 100ms delay [HERE](https://github.com/elastic/kibana/blob/v8.9.0/src/plugins/console/public/lib/autocomplete/autocomplete.ts#L1117), which is the reason why autocomplete does not appear without a short interval between operations.

So, I'm afraid there's no way to avoid using `sleep`. In addition to that, because I need to depend on `sleep` more to write tests, I split it out to iconic `sleepforDebouncePeriod(duration: number = 100)` function [HERE](https://github.com/elastic/kibana/pull/163233/files#diff-aae177e36732dbd9f33401541f7b6e9ede4482a4628bd3e85871201df6bc5184R590-R592).

Reply:
Furthermore, I increased the default sleep period to 200ms because lodash's debounce uses inaccurate setTimeout.

https://github.com/lodash/lodash/blob/4.17.21/lodash.js#L10488

Reply:
Tests for auto-complete have no longer been flaky since this build. 100ms delay after sync exec JS code seems to work well with lodash's 100ms debounce.
> 💔 [Build #151657](https://buildkite.com/elastic/kibana-pull-request/builds/151657) failed https://github.com/elastic/kibana/commit/5ec4ced9c5191c5be9ef7e7a0541f9c13b5213eb